### PR TITLE
add some missing rocksdb options

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed, added or deleted? (Required)

https://github.com/tikv/tikv/pull/14527

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

https://github.com/tikv/tikv/pull/14527

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tikv-configuration-file.md [hunk 1]
```diff
@@ -1344,6 +1344,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Default value for `defaultcf`: `true`
 + Default value for `writecf` and `lockcf`: `false`
 
+### `optimize-filters-for-memory`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1347]:
```suggestion
### `optimize-filters-for-memory` <span class="version-mark">New in v7.1.0</span>
```

File: tikv-configuration-file.md [hunk 2]
```diff
@@ -1361,6 +1367,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Determines whether each block creates a bloom filter
 + Default value: `false`
 
+### `ribbon-filter-above-level`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1370]:
```suggestion
### `ribbon-filter-above-level` <span class="version-mark">New in v7.1.0</span>
```

File: tikv-configuration-file.md [hunk 3]
```diff
@@ -1511,6 +1523,18 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Default value: `"128MB"`
 + Unit: KB|MB|GB
 
+### `format-version`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1526]:
```suggestion
### `format-version` <span class="version-mark">New in v6.5.0</span>
```

Please update the release-6.5.0.md:
https://github.com/pingcap/docs/blob/1db43f06719ac567cd848e2e8f8d7616c8a90a38/releases/release-6.5.0.md?plain=1#L340-L348

Comment (tabokie) [reply tikv-configuration-file.md]: There're plenty of new configurations before 7.1 that are not documented. Let's stick to the 7.1 issues for now.

File: tikv-configuration-file.md [hunk 4]
```diff
@@ -1344,6 +1344,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Default value for `defaultcf`: `true`
 + Default value for `writecf` and `lockcf`: `false`
 
+### `optimize-filters-for-memory` <span class="version-mark">New in v7.1.0</span>
+
++ Option to generate Bloom/Ribbon filters that minimize memory internal fragmentation.
++ Only effective for `format-version` >= 5.
++ Default value: `false`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1351]:
```suggestion
### `optimize-filters-for-memory` <span class="version-mark">New in v7.1.0</span>

+ Determines whether to generate Bloom/Ribbon filters that minimize memory internal fragmentation.
+ Note that this is only effective when [`format-version`](#format-version-new-in-v630) >= 5.
+ Default value: `false`
```

File: tikv-configuration-file.md [hunk 5]
```diff
@@ -1361,6 +1367,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Determines whether each block creates a bloom filter
 + Default value: `false`
 
+### `ribbon-filter-above-level` <span class="version-mark">New in v7.1.0</span>
+
++ Use Ribbon filter for levels higher or equal to this value. Use non-block-based bloom filter for lower levels. When this is set, `block-based-bloom-filter` will be ignored.
++ Only effective for `format-version` >= 5.
++ Default value: `false`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1374]:
```suggestion
+ Determins whether to use Ribbon filter for levels greater than or equal to this value. Use non-block-based bloom filter for lower levels. When this is set, [`block-based-bloom-filter`](#block-based-bloom-filter) will be ignored.
+ Note that this is only effective when [`format-version`](#format-version-new-in-v630) >= 5.
+ Default value: `false`
```

File: tikv-configuration-file.md [hunk 6]
```diff
@@ -1511,6 +1523,18 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
...
+
++ The format version of SST files. This option only affects newly written tables. When reading existing tables, the information about version is read from the footer.
++ Optional values:
+    - `0`: This version can be read by all TiKV releases. Doesn't support changing checksum type (default is CRC32).
+    - `1`:  Can be read by all TiKV releases. Supports non-default checksum, like xxHash. It is written by RocksDB when BlockBasedTableOptions::checksum is something other than kCRC32c. (version 0 is silently upconverted)
+    - `2`: Can be read by all TiKV releases. Changes the way we encode compressed blocks with LZ4, BZip2 and Zlib compression.
+    - `3`: Can be read by TiKV's versions since 2.1. Changes the way we encode the keys in index blocks.
+    - `4`: Can be read by TiKV's versions since 3.0. Changes the way we encode the values in index blocks.
+    - `5`: Can be read by TiKV's versions since 6.1. Full and partitioned filters use a generally faster and more accurate Bloom filter implementation, with a different schema.
++ Default value: `2`
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1536]:
```suggestion
+ The format version of SST files. This configuration item only affects newly written tables. For existing tables, the version information is read from the footer.
+ Optional values:
    - `0`: Can be read by all TiKV versions. The default checksum type is CRC32 and this version does not support changing the checksum type.
    - `1`: Can be read by all TiKV versions. Supports non-default checksum types like xxHash. RocksDB only writes data when the checksum type is not CRC32. (version `0` is automatically upgraded)
    - `2`: Can be read by all TiKV versions. Changes the encoding of compressed blocks using LZ4, BZip2 and Zlib compression.
    - `3`: Can be read by TiKV v2.1 and later versions. Changes the encoding of the keys in index blocks.
    - `4`: Can be read by TiKV v3.0 and later versions. Changes the encoding of the values in index blocks.
    - `5`: Can be read by TiKV v6.1 and later versions. Full and partitioned filters use a generally faster and more accurate Bloom filter implementation with a different schema.
+ Default value: `2`
```

File: tikv-configuration-file.md [hunk 7]
```diff
@@ -1511,6 +1523,18 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Default value: `"128MB"`
 + Unit: KB|MB|GB
 
+### `format-version` <span class="version-mark">New in v6.3.0</span>
```
Comment (Oreoxmt) [inline tikv-configuration-file.md:1526]:
```suggestion
### `format-version` <span class="version-mark">New in v6.2.0</span>
```
Keep it consistent with [releases/release-6.2.0.md](https://github.com/pingcap/docs/pull/13582/files#diff-d29bf2d43d9f4861e58933e6deea892bc12416c947be01e43ed742443c64f396R286)

**Review by Oreoxmt — APPROVED** [review]

File: releases/release-7.1.0.md [hunk 8]
```diff
@@ -253,6 +253,8 @@ In v7.1.0, the key new features and improvements are as follows:
...
 | TiDB | [`lite-init-stats`](/tidb-configuration-file.md#lite-init-stats-new-in-v710) | Newly added | Controls whether to use lightweight statistics initialization during TiDB startup. |
 | TiDB | [`timeout`](/tidb-configuration-file.md#timeout-new-in-v710) | Newly added | Sets the timeout for log-writing operations in TiDB. In case of a disk failure that prevents logs from being written, this configuration item can trigger the TiDB process to panic instead of hang. The default value is `0`, which means no timeout is set. |
+| TiKV | [`optimize-filters-for-memory`](/tikv-configuration-file.md#optimize-filters-for-memory-new-in-v710) | Newly added | Controls whether to generate Bloom/Ribbon filters that minimize memory internal fragmentation. |
+| TiKV | [`ribbon-filter-above-level`](/tikv-configuration-file.md#ribbon-filter-above-level-new-in-v710) | Newly added | Controls whether to use Ribbon filter for levels greater than or equal to this value and use non-block-based bloom filter for levels less than this value. |
```
Comment (hfxsd) [inline releases/release-7.1.0.md:257]:
```suggestion
| TiKV | [`ribbon-filter-above-level`](/tikv-configuration-file.md#ribbon-filter-above-level-new-in-v710) | Newly added | Controls whether to use Ribbon filters for levels greater than or equal to this value and use non-block-based bloom filters for levels less than this value. |
```

File: tikv-configuration-file.md [hunk 9]
```diff
@@ -1344,6 +1344,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Default value for `defaultcf`: `true`
 + Default value for `writecf` and `lockcf`: `false`
 
+### `optimize-filters-for-memory` <span class="version-mark">New in v7.1.0</span>
+
++ Determines whether to generate Bloom/Ribbon filters that minimize memory internal fragmentation.
++ Note that this is only effective when [`format-version`](#format-version-new-in-v620) >= 5.
```
Comment (hfxsd) [inline tikv-configuration-file.md:1350]:
```suggestion
+ Note that this parameter item takes effect only when [`format-version`](#format-version-new-in-v620) >= 5.
```

File: tikv-configuration-file.md [hunk 10]
```diff
@@ -1361,6 +1367,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Determines whether each block creates a bloom filter
 + Default value: `false`
 
+### `ribbon-filter-above-level` <span class="version-mark">New in v7.1.0</span>
+
++ Determines whether to use Ribbon filter for levels greater than or equal to this value and use non-block-based bloom filter for levels less than this value. When this is set, [`block-based-bloom-filter`](#block-based-bloom-filter) will be ignored.
++ Note that this is only effective when [`format-version`](#format-version-new-in-v620) >= 5.
```
Comment (hfxsd) [inline tikv-configuration-file.md:1373]:
```suggestion
+ Note that this configuration item takes effect only when [`format-version`](#format-version-new-in-v620) >= 5.
```

File: tikv-configuration-file.md [hunk 11]
```diff
@@ -1361,6 +1367,12 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
 + Determines whether each block creates a bloom filter
 + Default value: `false`
 
+### `ribbon-filter-above-level` <span class="version-mark">New in v7.1.0</span>
+
++ Determines whether to use Ribbon filter for levels greater than or equal to this value and use non-block-based bloom filter for levels less than this value. When this is set, [`block-based-bloom-filter`](#block-based-bloom-filter) will be ignored.
```
Comment (hfxsd) [inline tikv-configuration-file.md:1372]:
```suggestion
+ Determines whether to use Ribbon filters for levels greater than or equal to this value and use non-block-based bloom filters for levels less than this value. When this configuration item is set, [`block-based-bloom-filter`](#block-based-bloom-filter) will be ignored.
```

File: tikv-configuration-file.md [hunk 12]
```diff
@@ -1511,6 +1523,18 @@ Configuration items related to `rocksdb.defaultcf`, `rocksdb.writecf`, and `rock
...
+### `format-version` <span class="version-mark">New in v6.2.0</span>
+
++ The format version of SST files. This configuration item only affects newly written tables. For existing tables, the version information is read from the footer.
++ Optional values:
+    - `0`: Can be read by all TiKV versions. The default checksum type is CRC32 and this version does not support changing the checksum type.
+    - `1`: Can be read by all TiKV versions. Supports non-default checksum types like xxHash. RocksDB only writes data when the checksum type is not CRC32. (version `0` is automatically upgraded)
+    - `2`: Can be read by all TiKV versions. Changes the encoding of compressed blocks using LZ4, BZip2 and Zlib compression.
+    - `3`: Can be read by TiKV v2.1 and later versions. Changes the encoding of the keys in index blocks.
+    - `4`: Can be read by TiKV v3.0 and later versions. Changes the encoding of the values in index blocks.
+    - `5`: Can be read by TiKV v6.1 and later versions. Full and partitioned filters use a generally faster and more accurate Bloom filter implementation with a different schema.
```
Comment (hfxsd) [inline tikv-configuration-file.md:1535]:
```suggestion
    - `5`: Can be read by TiKV v6.1 and later versions. Full and partitioned filters use a faster and more accurate Bloom filter implementation with a different schema.
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**